src/app-logic/url-handling.js

```diff
@@ -594,6 +594,66 @@ export function stateFromLocation(
   };
 }
 
+// Converts "7_10-6-2_5" to [7, 8, 9, 10, 6, 2, 3, 4, 5]
+function convertUintListFromString(s: string): number[] {
+  const pieces = s.split('-');
+  const numbers = [];
+  for (const piece of pieces) {
+    if (piece.indexOf('_') !== -1) {
+      const [fromStr, toInclusiveStr] = piece.split('_');
+      const from = +fromStr;
+      const toInclusive = +toInclusiveStr;
+      if (!isNaN(from) && !isNaN(toInclusive)) {
+        for (let i = from; i <= toInclusive; i++) {
+          numbers.push(i);
+        }
+      } else {
+        numbers.push(NaN);
+      }
+    } else {
+      numbers.push(+piece);
+    }
+  }
+  return numbers;
+}
+
+// Converts [7, 8, 9, 10, 6, 2, 3, 4, 5] to "7_10-6-2_5"
+function convertUintListToString(numbers: number[]): string {
+  const pieces = [];
+  for (let i = 0; i < numbers.length; i++) {
+    const number = numbers[i];
+    if (i === numbers.length - 1 || numbers[i + 1] !== number + 1) {
+      pieces.push(`${number}`);
```

Comment:
nit: why don't you push `number` directly?

Reply:
Hmm, it felt cleaner to have `pieces` be of type `string[]` rather than a mixed array of strings and numbers.